src/modules/app/tasks/tasks.component.ts

```diff
@@ -44,16 +79,105 @@ export class TasksComponent implements OnInit {
   public addTask(task: Task): void {
     this.fbTasksService.createTaskInDb(task, JSON.parse(localStorage.uid)).subscribe(task => {
       this.tasks.push(task);
+      this.newTasks.push(task);
     }, err => {
       this.error = err.message;
     })
   }
 
-  public deleteTask(task: Task): void {
-    this.fbTasksService.deleteTaskInDb(task, JSON.parse(localStorage.uid)).subscribe(() => {
-      this.tasks = this.tasks.filter(t => t.id !== task.id);
+  public preDelTask(task: Task): void {
+    this.showDelTaskModal = true;
+    this.currentDelTask = task;
+    this.disableScroll();
+  }
+
+  public deleteTask(): void {
+    this.fbTasksService.deleteTaskInDb(this.currentDelTask, JSON.parse(localStorage.uid)).subscribe(() => {
+      this.tasks = this.tasks.filter(t => t.id !== this.currentDelTask.id);
+      this.newTasks = this.newTasks.filter(t => t.id !== this.currentDelTask.id);
+      this.inProcessTasks = this.inProcessTasks.filter(t => t.id !== this.currentDelTask.id);
+      this.doneTasks = this.doneTasks.filter(t => t.id !== this.currentDelTask.id);
+      this.hideDelModal();
     }, err => {
       this.error = err.message;
+      this.hideDelModal();
     })
   }
+
+  public disableScroll(): void {
+    const scrollWidth = window.innerWidth - document.body.offsetWidth;
+    this.topWindowScroll = window.scrollY;
+    document.body.style.cssText = `
+        position: fixed;
+        top: ${-window.scrollY}px;
+        left: 0;
+        width: 100%;
+        height: 100vh;
+        overflow: hidden;
+        padding-right: ${scrollWidth}px;
+    `;
+  }
```

Comment:
what is that?

Reply:
This method adds padding to the browser window and when the modal window is opened, the browser does not move.